conans/client/generators/cmake_find_package.py

```diff
@@ -229,38 +223,16 @@ def content(self):
             )
         return ret
 
-    def _get_components(self, pkg_name, pkg_findname, cpp_info):
-        find_package_components = []
-        for comp_name, comp in self.sorted_components(cpp_info).items():
-            comp_findname = self._get_name(cpp_info.components[comp_name])
+    def _get_components(self, pkg_name, cpp_info):
+        generator_components = super(CMakeFindPackageGenerator, self)._get_components(pkg_name,
+                                                                                      cpp_info)
+        ret = []
+        for comp_genname, comp, comp_requires_gennames in generator_components:
             deps_cpp_cmake = DepsCppCmake(comp)
-            deps_cpp_cmake.public_deps = self._get_component_requires(pkg_name, pkg_findname, comp)
-            find_package_components.append((comp_findname, deps_cpp_cmake))
-        find_package_components.reverse()  # From the less dependent to most one
-        return find_package_components
-
-    def _get_component_requires(self, pkg_name, pkg_findname, comp):
-        comp_requires_findnames = []
-        for require in comp.requires:
-            if COMPONENT_SCOPE in require:
-                comp_require_pkg_name, comp_require_comp_name = require.split(COMPONENT_SCOPE)
-                comp_require_pkg = self.deps_build_info[comp_require_pkg_name]
-                comp_require_pkg_findname = self._get_name(comp_require_pkg)
```

Comment:
Is it possible that this refactor hides a bug?
- In the previous implementation, the ``comp_require_pkg_findname`` is different when it is in another package. 
- The new implementation seems to use always the same.

Can this behavior be checked with a test to show that it was a bug in the past? Or am I missing something? 
I am finding this refactor is a bit difficult to understand while reading it that it is not changing behavior.

Reply:
The new implementation takes into account if the component comes from a different package or from the same one:

```python
pkg, cmp = req.split(COMPONENT_SCOPE) if COMPONENT_SCOPE in req else (pkg_name, req)
```

The `pkg` variable uses the one from the requires (if `req` is something like `pkg::cmp`) or the one from the same package (if `req` is something like `cmp`).

It was not a bug and it is not a bug. We need to use the package that provides the component and call `_get_name`  with it. IMO it doesn't require any extra test, there are already lot of tests using components and it is very hard to have a false positive here: it'd require a package with that name in `deps_build_info` providing a component `deps_build_info[pkg_name].components[cmp_name]` and returning the expected value.